src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js

```diff
@@ -1874,4 +1878,129 @@ describe('ReactDOMComponent', () => {
       expect(container.firstChild.innerHTML).toBe(html2);
     });
   });
+
+  describe('Attributes with aliases', function() {
+    it('does not set aliased attributes on DOM elements', function() {
+      spyOn(console, 'error');
+
+      var el = ReactTestUtils.renderIntoDocument(<div class="test" />);
+
+      expect(el.className).toBe('');
+
+      expectDev(console.error.calls.argsFor(0)[0]).toContain(
+        'Warning: Unknown DOM property class. Did you mean className?',
+      );
+    });
+
+    it('does not set aliased attributes on SVG elements', function() {
+      spyOn(console, 'error');
+
+      var el = ReactTestUtils.renderIntoDocument(
+        <svg><text arabic-form="initial" /></svg>,
+      );
+      var text = el.querySelector('text');
+
+      expect(text.getAttribute('arabic-form')).toBe(null);
+
+      expectDev(console.error.calls.argsFor(0)[0]).toContain(
+        'Warning: Unknown DOM property arabic-form. Did you mean arabicForm?',
+      );
+    });
+
+    it('sets aliased attributes on custom elements', function() {
+      spyOn(console, 'error');
+
+      var el = ReactTestUtils.renderIntoDocument(
+        <div is="custom-element" class="test" />,
+      );
+
+      expect(el.getAttribute('class')).toBe('test');
+
+      expectDev(console.error).not.toHaveBeenCalled();
+    });
+
+    it('updates aliased attributes on custom elements', function() {
+      var container = document.createElement('div');
+
+      spyOn(console, 'error');
+
```

Comment:
Yep: e68bf4d. I tried to match up to other tests, but I might have missed the mark. Looks like clumps of work with a line between that block and the assertion?